[it] Localization function-as-a-service.md (#1630)
cncf/glossary: content/it/function-as-a-service.md

Il data center deve essere costruito per soddisfare i picchi di domanda, anche quando i carichi di lavoro diminuiscono e quelle risorse restano inattive.
Viceversa, se l'azienda cresce rapidamente, il reparto IT potrebbe avere difficoltà a tenere il passo.
In base a un modello di cloud computing standard [Infrastructure-as-a-Service (IaaS)](/it/infrastructure-as-a-service/),
gli utenti preacquistano unità di capacità, il che significa pagare un provider di cloud pubblico per componenti server sempre attivi su cui eseguire le proprie app.
È responsabilità dell'utente aumentare la capacità del server durante i periodi di forte domanda
e ridimensionare quando tale capacità non è più necessaria.
Insomma, l'infrastruttura cloud necessaria per eseguire un'app rimane attiva anche quando l'app non viene utilizzata.

## In che modo aiuta

FaaS offre agli sviluppatori un'[astrazione](/it/abstraction/) per l'esecuzione di applicazioni Web in risposta a eventi senza gestire i server.
Ad esempio, il caricamento di un file potrebbe attivare un codice personalizzato che transcodifica il file in vari formati.
L'infrastruttura FaaS ridimensionerà automaticamente il codice per un uso intenso,

> e lo sviluppatore non dovrà dedicare tempo o risorse alla creazione del codice per garantirne la [scalabilità](/it/scalabilità/).

Reviewer: ```suggestion
e lo sviluppatore non dovrà dedicare tempo o risorse alla creazione del codice per garantirne la [scalabilità](/it/scalability/).
```

---
title: Function as a Service (FaaS)
status: Completed
category: Tecnologia
tags: ["infrastruttura", "", ""]
---

## Cos'è

> Function as a Service (FaaS) è un tipo di [servizio](/service/) di [cloud computing](it/cloud-computing/) [serverless](it/serverless/) 

Reviewer 2: ```suggestion
Function as a Service (FaaS) è un tipo di [servizio](/it/service/) di [cloud computing](/it/cloud-computing/) [serverless](/it/serverless/) 
```

che consente l'esecuzione di codice in risposta agli eventi senza mantenere la complessa infrastruttura

> tipicamente associata alla creazione e all'avvio di applicazioni a [microservizi](it/microservices/).

Reviewer 2: ```suggestion
tipicamente associata alla creazione e all'avvio di applicazioni a [microservizi](/it/microservices/).
```

Con FaaS, gli utenti gestiscono solo funzioni e dati mentre il provider cloud gestisce l'applicazione.
Ciò consente agli sviluppatori di ottenere le funzioni di cui hanno bisogno senza pagare per i servizi quando il codice non è in esecuzione.

> Alcuni esempi FaaS popolari includono: [AWS Lambda di Amazon](https://aws.amazon.com/it/lambda/),

Reviewer 3: è stato rimosso dall'inglese, nel frattempo
```suggestion
```

> [Google Cloud Functions](https://cloud.google.com/functions/) e [icrosoft Azure Functions](https://azure.microsoft.com/it-it/products/functions/).

Reviewer 2: ```suggestion
[Google Cloud Functions](https://cloud.google.com/functions/) e [Microsoft Azure Functions](https://azure.microsoft.com/it-it/products/functions/).
```

> [Google Cloud Functions](https://cloud.google.com/functions/) e [Microsoft Azure Functions](https://azure.microsoft.com/it-it/products/functions/).

Reviewer 3: è stato rimosso dall'inglese, nel frattempo
```suggestion
```

 

## Quale problema affronta

In uno scenario tradizionale on-premise, un'azienda gestisce e mantiene il proprio data center.
L'azienda deve investire in server, storage, software e altre tecnologie

> e potenzialmente assumere uno staff IT o appaltare l'acquisto, la gestione e l'aggiornamento di tutte le apparecchiature e le licenze.

Reviewer 3: ```suggestion
e potenzialmente assumere uno staff IT o dei consulenti per l'acquisto, la gestione e l'aggiornamento di tutte le apparecchiature e le licenze.
```
con riferimento al fatto che di queste attivita se ne devono occupare o lo staff IT o i consulenti.

Il data center deve essere costruito per soddisfare i picchi di domanda, anche quando i carichi di lavoro diminuiscono e quelle risorse restano inattive.

> Vice versa, se l'azienda cresce rapidamente, il reparto IT potrebbe avere difficoltà a tenere il passo.

Reviewer 2: ```suggestion
Viceversa, se l'azienda cresce rapidamente, il reparto IT potrebbe avere difficoltà a tenere il passo.
```

> In base a un modello di cloud computing standard [Infrastructure-as-a-Service (IaaS)](it/infrastructure-as-a-service/),

Reviewer 2: ```suggestion
In base a un modello di cloud computing standard [Infrastructure-as-a-Service (IaaS)](/it/infrastructure-as-a-service/),
```

gli utenti preacquistano unità di capacità, il che significa pagare un provider di cloud pubblico per componenti server sempre attivi su cui eseguire le proprie app.
È responsabilità dell'utente aumentare la capacità del server durante i periodi di forte domanda
e ridimensionare quando tale capacità non è più necessaria.
Insomma, l'infrastruttura cloud necessaria per eseguire un'app rimane attiva anche quando l'app non viene utilizzata.

## In che modo aiuta

> FaaS offre agli sviluppatori un'[astrazione](it/abstraction/) per l'esecuzione di applicazioni Web in risposta a eventi senza gestire i server.

Reviewer 2: ```suggestion
FaaS offre agli sviluppatori un'[astrazione](/it/abstraction/) per l'esecuzione di applicazioni Web in risposta a eventi senza gestire i server.
```

Ad esempio, il caricamento di un file potrebbe attivare un codice personalizzato che transcodifica il file in vari formati.
L'infrastruttura FaaS ridimensionerà automaticamente il codice per un uso intenso,

> e lo sviluppatore non dovrà dedicare tempo o risorse alla creazione del codice per garantirne la [scalabilità](it/scalabilità/).

Reviewer 2: ```suggestion
e lo sviluppatore non dovrà dedicare tempo o risorse alla creazione del codice per garantirne la [scalabilità](/it/scalabilità/).
```

> e lo sviluppatore non dovrà dedicare tempo o risorse alla creazione del codice per garantirne la [scalabilità](/it/scalabilità/).

Reviewer 3: qui è più bella la traduzione dell'originale, seriamente!! Brava @annalisag-spark !!!
